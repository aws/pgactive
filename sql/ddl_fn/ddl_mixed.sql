-- test for RT-#37869
SELECT bdr.bdr_replicate_ddl_command($DDL$ 
CREATE TABLE public.add_column (
    id serial primary key,
    data text
);
$DDL$);

INSERT INTO add_column (data) SELECT generate_series(1,100,10);

SELECT bdr.bdr_replicate_ddl_command($DDL$ ALTER TABLE public.add_column ADD COLUMN other varchar(100); $DDL$);

SELECT bdr.bdr_wait_for_slots_confirmed_flush_lsn(NULL,NULL);
\c postgres
SELECT id, data, other FROM add_column ORDER BY id;

UPDATE add_column SET other = 'foobar'; 

SELECT bdr.bdr_wait_for_slots_confirmed_flush_lsn(NULL,NULL);
\c regression
SELECT id, data, other FROM add_column ORDER BY id;

SELECT bdr.bdr_replicate_ddl_command($DDL$ 
DROP TABLE public.add_column
$DDL$);

-- We allow BDR nodes sending out changes for postgres logical replication
-- subscribers.
SELECT bdr.bdr_replicate_ddl_command($DDL$
CREATE PUBLICATION mypub FOR ALL TABLES;
$DDL$);

-- We do not allow BDR nodes receiving changes from postgres logical
-- replication publishers.
SELECT bdr.bdr_replicate_ddl_command($DDL$
CREATE SUBSCRIPTION mysub CONNECTION '' PUBLICATION mypub;
$DDL$);

SELECT bdr.bdr_replicate_ddl_command($DDL$
ALTER SUBSCRIPTION mysub REFRESH PUBLICATION;
$DDL$);

SELECT bdr.bdr_replicate_ddl_command($DDL$
DROP PUBLICATION mypub;
$DDL$);

-- We do not allow external logical replication extensions to be created when
-- BDR is active.
SELECT bdr.bdr_replicate_ddl_command($DDL$
CREATE EXTENSION pglogical;
$DDL$);
