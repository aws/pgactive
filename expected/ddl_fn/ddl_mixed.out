-- test for RT-#37869
SELECT bdr.bdr_replicate_ddl_command($DDL$ 
CREATE TABLE public.add_column (
    id serial primary key,
    data text
);
$DDL$);
 bdr_replicate_ddl_command 
---------------------------
 
(1 row)

INSERT INTO add_column (data) SELECT generate_series(1,100,10);
SELECT bdr.bdr_replicate_ddl_command($DDL$ ALTER TABLE public.add_column ADD COLUMN other varchar(100); $DDL$);
 bdr_replicate_ddl_command 
---------------------------
 
(1 row)

SELECT bdr.bdr_wait_for_slots_confirmed_flush_lsn(NULL,NULL);
 bdr_wait_for_slots_confirmed_flush_lsn 
----------------------------------------
 
(1 row)

\c postgres
SELECT id, data, other FROM add_column ORDER BY id;
 id | data | other 
----+------+-------
  1 | 1    | 
  2 | 11   | 
  3 | 21   | 
  4 | 31   | 
  5 | 41   | 
  6 | 51   | 
  7 | 61   | 
  8 | 71   | 
  9 | 81   | 
 10 | 91   | 
(10 rows)

UPDATE add_column SET other = 'foobar'; 
SELECT bdr.bdr_wait_for_slots_confirmed_flush_lsn(NULL,NULL);
 bdr_wait_for_slots_confirmed_flush_lsn 
----------------------------------------
 
(1 row)

\c regression
SELECT id, data, other FROM add_column ORDER BY id;
 id | data | other  
----+------+--------
  1 | 1    | foobar
  2 | 11   | foobar
  3 | 21   | foobar
  4 | 31   | foobar
  5 | 41   | foobar
  6 | 51   | foobar
  7 | 61   | foobar
  8 | 71   | foobar
  9 | 81   | foobar
 10 | 91   | foobar
(10 rows)

SELECT bdr.bdr_replicate_ddl_command($DDL$ 
DROP TABLE public.add_column
$DDL$);
 bdr_replicate_ddl_command 
---------------------------
 
(1 row)

-- We allow BDR nodes sending out changes for postgres logical replication
-- subscribers.
SELECT bdr.bdr_replicate_ddl_command($DDL$
CREATE PUBLICATION mypub FOR ALL TABLES;
$DDL$);
 bdr_replicate_ddl_command 
---------------------------
 
(1 row)

-- We do not allow BDR nodes receiving changes from postgres logical
-- replication publishers.
SELECT bdr.bdr_replicate_ddl_command($DDL$
CREATE SUBSCRIPTION mysub CONNECTION '' PUBLICATION mypub;
$DDL$);
ERROR:  CREATE SUBSCRIPTION is not supported when bdr is active
CONTEXT:  during DDL replay of ddl statement: 
CREATE SUBSCRIPTION mysub CONNECTION '' PUBLICATION mypub;

SELECT bdr.bdr_replicate_ddl_command($DDL$
ALTER SUBSCRIPTION mysub REFRESH PUBLICATION;
$DDL$);
ERROR:  ALTER SUBSCRIPTION is not supported when bdr is active
CONTEXT:  during DDL replay of ddl statement: 
ALTER SUBSCRIPTION mysub REFRESH PUBLICATION;

SELECT bdr.bdr_replicate_ddl_command($DDL$
DROP PUBLICATION mypub;
$DDL$);
 bdr_replicate_ddl_command 
---------------------------
 
(1 row)

-- We do not allow external logical replication extensions to be created when
-- BDR is active.
SELECT bdr.bdr_replicate_ddl_command($DDL$
CREATE EXTENSION pglogical;
$DDL$);
ERROR:  cannot create an external logical replication extension when BDR is active
CONTEXT:  during DDL replay of ddl statement: 
CREATE EXTENSION pglogical;

